pyproject.toml

```diff
@@ -20,6 +20,7 @@ dependencies = [
     "pyarrow",
     "deprecated",
     "ipykernel", # Support for Jupyter notebooks
+    "scikit-learn",
```

Comment:
Can we start a new optional dependencies section for this kind of thing? I know that for other projects, having a hard dependency on scikit-learn has led to issues (and it would be good to set precedent now before dependency bloat sets in).

Reply:
Sure, that sounds reasonable, though it is definitely necessary for the current crossmatch algorithm to work

Reply:
I think this is probably something that should be changed in PPT though, it'll need changes to the GitHub Actions to make sure the optional dependencies are installed for unit tests

Reply:
Added https://github.com/lincc-frameworks/python-project-template/issues/270